pandas/core/strings/accessor.py

```diff
@@ -1274,14 +1274,14 @@ def contains(
         1     True
         2    False
         3    False
-        4      NaN
-        dtype: object
+        4    False
+        dtype: bool
 
         Returning an Index of booleans using only a literal pattern.
 
         >>> ind = pd.Index(["Mouse", "dog", "house and parrot", "23.0", np.nan])
         >>> ind.str.contains("23", regex=False)
-        Index([False, False, False, True, nan], dtype='object')
+        array([False, False, False,  True, False])
```

Comment:
The fact that it changes here from Index to array does not seem to be intentional, though. Will look into that.

Reply:
See the links I mentioned above in https://github.com/pandas-dev/pandas/pull/61908#issuecomment-3092574633. 
The main reason is that the current "object-dtype with NaN" is not that useful in practice. For example, it means that boolean filtering like `ser[ser.str.contains("B")]` only works as long as `ser` does not contain missing values